syntax = "proto3";

package emailverifier;

service EmailVerifier {
  rpc CheckEmail (EmailRequest) returns (EmailReply) {}
}

message EmailRequest {
  string email = 1;
}

message EmailReply {
  bool is_valid = 1;
  string domain = 2;
  bool is_common_domain = 3;
}
